Stack/Opc.Ua.Core/Stack/Configuration/EndpointDescription.cs

```diff
@@ -124,8 +125,8 @@ public UserTokenPolicy FindUserTokenPolicy(UserTokenType tokenType, string issue
             // construct issuer type.
             string issuedTokenTypeText = issuedTokenType;
 
-            // find matching policy.
-            foreach (UserTokenPolicy policy in m_userIdentityTokens)
+            // find matching policy, return most secure matching policy first.
+            foreach (UserTokenPolicy policy in m_userIdentityTokens.OrderByDescending(token => SecuredApplication.CalculateSecurityLevel(MessageSecurityMode.Sign, token.SecurityPolicyUri)))
```

Comment:
would it make sense to Order these when the property is set or initialized, to avoid sort every time?

Reply:
We expose public "Add" and "Remove" Functions with the property, therefore I see this as the safer solution. Unless we make this a readonly Collection

Reply:
thats about right...